controller/lib/pid/pid.cpp

```diff
@@ -64,12 +62,20 @@ bool PID::Compute() {
     return false;
   unsigned long now = Hal.millis();
   unsigned long timeChange = (now - lastTime);
-  if (timeChange >= SampleTime) {
+  // compute actual samples time-difference to take jitter into account in
+  // integral and derivative
+  double samplesTimeChangeSec = ((now - lastSampleTime) / 1000.0);
+  // condition to update output : 1 sample time has passed and we have new data
+  if (timeChange >= SampleTime && samplesTimeChangeSec > 0) {
     /*Compute all the working error variables*/
     double input = *myInput;
     double error = *mySetpoint - input;
-    double dInput = (input - lastInput);
-    outputSum += (ki * error);
+    double dInput = 0.0;
+    // Compute dInput only if needed (P_ON_M or D_ON_M)
```

Comment:
Per code below, dInput is actually only needed at line 97, can we just declare it there?

Reply:
this means changing the comment "compute all the working variables", but sure ;)

Reply:
I take that back, it is also required on line 81 for proportional on measurement mode, which I think is why I went that route in the first place